### DISCUSSION TITLE
GSOC 2024 Idea : Improve  User Experience and  Maintainability of IHR Django

### DISCUSSION THREAD

**Discussion by Forchapeatl** [InternetHealthReport/gsoc#40] [Ideas]
**Project:** Improve User Experience and Maintainability of IHR Django:

**Brief explanation:**
As I grow deeper in passion for internet health. I have realized a few bugs, slow response time, Gateway Timeouts  and negligence on the current ihr-django project. The readme file neglates relative paths and introduces  docker when basic django installation steps have not been completed.  There are several bugs encountered upon successful installation of the ihr-django project which I have listed below. The goal of this project is to create a more user friendly experience  for both developers and users

**Expected Results:**

1. Update IHR  django project from 2.2.27 to 4.2.0 :
 This will lead to compatibility with newer technologies,  reduced memory usage,support for Django Channels and websockets events.

2. Asynchronous processing :
 For long-running tasks or resource-intensive operations, explore implementing asynchronous processing on global alarms report,  to prevent blocking the main thread and improve responsiveness. 

3. Improve documentation  and user experience:

- -Develop interactive tutorials or guides to walk users through using the API for common tasks.
- -Provide sample db.sqlite for newcomers to play with.

4. Sanitize SQL injections:
Implement proper input validation to prevent malicious attacks like SQL injection or cross-site scripting (XSS).
Sanitize user input to remove any potentially harmful characters before processing them

5. Gather user feedback Form:
By understanding what information users find valuable, what features they use most, and what aspects of the report are confusing, IHR can prioritize improvements and optimize the website and user experience.

6. Delete production migration files from the github repository:

**Knowledge Prerequisite:**  Django , VueJS , Docker , PostgreSQL

**Project size**: 350 hours

**Resources:**
[Incompatible versions for requests and charset-normalizer #57](https://github.com/InternetHealthReport/ihr-django/issues/57)
[[DOC] Make installation Docs more precise #55](https://github.com/InternetHealthReport/ihr-django/issues/55)
[Reverse for 'discoGeoData' not found. 'discoGeoData' is not a valid view function or pattern name](https://github.com/InternetHealthReport/ihr-django/issues/60)
[Reverse for 'discoDetail' not found. 'discoDetail' is not a valid view function or pattern name.](https://github.com/InternetHealthReport/ihr-django/issues/61)

**Comment by Forchapeatl** [comment]
@mohamedawnallah , @romain-fontugne  and @mhmdahmedfathi please for your kind reviews on this new project idea.

**Reply by romain-fontugne** [reply]
thanks, updating/maintaining better that repository is indeed needed. I think allocating some time to update the production server would be also required

**Reply by dpgiakatos** [reply]
Actually, we do not need Django. We should replace it with FastAPI, Falcon API, or Flask. Additionally, it would be a good opportunity to refactor the database.

**Reply by Forchapeatl** [reply]
@dpgiakatos , this means the ihr-django repository will be completely depreciated. Now I am so confused . I am afraid this great idea is coming up late and @romain-fontugne seems to be very busy .

**Reply by Forchapeatl** [reply]
@romain-fontugne , please for your take on this ?

**Reply by romain-fontugne** [reply]
I wouldn't say we 'should replace' Django but 'we could replace' it, especially if we need a lot of work to update it. We have to carefully weight the pros and cons of going either way.
